server/src/main/java/org/elasticsearch/action/explain/TransportExplainAction.java

```diff
@@ -72,19 +77,29 @@ public TransportExplainAction(
             threadPool.executor(ThreadPool.Names.GET)
         );
         this.searchService = searchService;
+        this.remoteClusterService = transportService.getRemoteClusterService();
     }
 
     @Override
     protected void doExecute(Task task, ExplainRequest request, ActionListener<ExplainResponse> listener) {
         request.nowInMillis = System.currentTimeMillis();
+
+        // Use a supplier to create the ResolvedIndices instance as necessary since it is only used to build the index metadata map for
+        // the query rewrite context
+        Supplier<ResolvedIndices> resolvedIndicesSupplier = createResolvedIndicesSupplier(request);
```

Comment:
I'm still using a supplier here because the `ResolvedIndices` instance is only used (lazily) by the query rewrite context, so most of the time it's not necessary to create it. LMK if this supplier usage is a problem.